students/Hovhannes_Muradyan/Lesson1/hw1.txt

```diff
@@ -0,0 +1,99 @@
+Задача 1.
+С помощью битовых операций проверить число на четность 
+примеры 7 -> false, 4578 -> true
+bool is_even(int x)
+{
+	return !(x&1);
+}
+
+
+
+
+
+
+
+Задача 2.
+Посчитать количество нулей в начале двоичного представления:
+пример x = 784252715 (00101110101111101011111100101011) -> функция должна возвращать 2
+int nulls_count(int x)
+{
+  int counter = 0;
+  for(int i = 30; i >=0 ; i--){
+    if(x & (1<<i)){
+      return counter;
+    }
+    counter++;
+  }
+}
+
+
+
+
+
+
+Задача 3.
+Даны числа x и i, требуется обнулить все биты числа x от самого старшeго до i.
+Пример x = 784252715 (00101110101111101011111100101011), i = 13 -> функция должна возвращать 442155 (00000000000000000001111100101011)
+# В примере ошибка , вместо 442155 должен быть 7979
+int clear_bits_through_i(int x, int i)
+{
+  int a = INT32_MAX;
+  a = a >> (31 - i);
+  x = x & a;
+  return x;
+}
+
+
+
+
+Задача 4.
+Пусть дано целочисленное значение short x, требуется возвращать его бинарное представление в std::string (порядок - big endian).
+Алгоритм должен использовать битовые операции.
+Примеры
+0   -> "0000000000000000"
+1   -> "0000000000000001"
+7   -> "0000000000000111"
+777 -> "0000001100001001"
+513 -> "0000001000000001"
+-7  -> "1111111111111001"
+
+string short_to_binary_string(short x)
+{
+  string k = "";
+  for(int i = 31; i >=0;i--){
```

Comment:
The solution is right (again change 31 by sizeof(short)*8-1, or maybe sizeof(x)*8-1 would be better, since x later can be changed to another type like int, unsigned or something else). 
Just a few remarks related to the C++ code, here you can skip the initialization of k, since it's by default an empty string (I mean `string k = ""`; is equivalent to `string k`;).
In C++ standard library, if you include `#include <sstream>` there is a `stringstream` class which is used for these kind of cases, when you need to construct a string (from ints, chars, etc.). Just to know that class you can test it by yourself and construct your string like this:   
```
stringstream ss;
for (int i = sizeof(x)*8 - 1; i >= 0; i--) {
    if (x & (1 << i)) {
        ss << 1;
    }
    else
        ss << 0;
}
return ss.str();
```
From C++ point if view stringstream is nicer, but of course your solution is also correct.

Just another solution that I used, you can have a look at it too, as an alternative solution:
```
string short_to_binary_string(short x)
{
	string s;
	for (int i = 0; i < sizeof(x) * 8; ++i)
	{
		s = to_string((x & 1) == 1) + s;
		x = x >> 1;
	}
	return s;
}
```